### PR TITLE
docs: added en version of udf functions development guideline

### PR DESCRIPTION
**Pull request by michelle-qinqin**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
added en version of udf functions development guideline
added the function statement docs into the relating index.rst file

### CODE REVIEW

File: docs/en/developer/udf_develop_guide.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# UDF Function Development Guideline
+## 1. Background
+Although there already are hundreds of built-in functions, they can not satisfy the needs in some cases. In the past, this could only be done by developing new built-in functions. Built-in function development requires a relatively long cycle because it needs to recompile binary files and wait for new version release.
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md]: Although there are already ...

Comment (lumianph) [reply docs/en/developer/udf_develop_guide.md]: and wait for new version release.
and users have to wait for new version release

File: docs/en/developer/udf_develop_guide.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+In order to help users quickly and flexibly resolve specific feature computing requirements, we develop the mechanism of user dynamic registration function.
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md]: In order to help users to quickly develop computing functions that are not provided by OpenMLDB

File: docs/en/developer/udf_develop_guide.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+SQL functions can be divided into single-line functions and aggregate functions. At present, only the custom development of single-line functions is supported. An introduction to single-line functions and aggregate functions can be seen [here](./built_in_function_develop_guide.md).
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md]: can be divided into
can be categorised into

Comment (lumianph) [reply docs/en/developer/udf_develop_guide.md]: only the custom development of single-line functions 
only single-line UDFs are supported

File: docs/en/developer/udf_develop_guide.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+| INT       | `int32_t`   |
+| BIGINT    | `int64_t`   |
+| FLOAT     | `float`     |
+| DOUBLE    | `double`    |
+| STRING    | `StringRef` |
+| TIMESTAMP | `Timestamp` |
+| DATE      | `Date`      |
+
+
+#### 2.1.3 函数参数和返回值
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md]: Parameters and Return Values

File: docs/en/developer/udf_develop_guide.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+**Return Value**:
+
+* If the output type of the UDF is basic type, it will be processed as return value.
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md]: a basic type
as a return value

File: docs/en/developer/udf_develop_guide.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+* If the output type of the UDF is STRING, TIMESTAMP or DATE, it will return through the last parameter of the function.
+
+**Parameters**: 
+
+* If the parameter is basic type, it will be passed as value.
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md]: a basic type
passed by value

File: docs/en/developer/udf_develop_guide.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+**Parameters**: 
+
+* If the parameter is basic type, it will be passed as value. 
+* If the output type of the UDF is STRING, TIMESTAMP or DATE, it will be passed by pointer. 
+* The first parameter must be `UDFContext* ctx`. The definition of [UDFContext](../../../include/udf/openmldb_udf.h) is:
+
+```c++
+    struct UDFContext {
+        ByteMemoryPool* pool;  // 用来分配内存
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md]: translate the Chinese comments in the entire doc

File: docs/en/developer/udf_develop_guide.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+- Copy the `include` directory (https://github.com/4paradigm/OpenMLDB/tree/main/include) to a certain path (like `/work/OpenMLDB/`) for later compiling. 
+- Run the compiling command. `-I` specifies the path of `include` directory. `-o` specifies the name of the dynamic library.
+
+```shell
+g++ -shared -o libtest_udf.so examples/test_udf.cc -I /work/OpenMLDB/include -std=c++11 -fPIC
+```
+
+### 2.3 Copy the Dynamic Library
+- The compiled dynamic libraries should be copied into the `udf` directory of the path which OpenMLDB `tablet/taskmanager` is deployed. Please create one if this directory does not exist.
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md]: which => where

File: docs/en/developer/udf_develop_guide.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+- Copy the `include` directory (https://github.com/4paradigm/OpenMLDB/tree/main/include) to a certain path (like `/work/OpenMLDB/`) for later compiling. 
+- Run the compiling command. `-I` specifies the path of `include` directory. `-o` specifies the name of the dynamic library.
+
+```shell
+g++ -shared -o libtest_udf.so examples/test_udf.cc -I /work/OpenMLDB/include -std=c++11 -fPIC
+```
+
+### 2.3 Copy the Dynamic Library
+- The compiled dynamic libraries should be copied into the `udf` directory of the path which OpenMLDB `tablet/taskmanager` is deployed. Please create one if this directory does not exist.  
+- Please note that the `udf` directory of `tablet` is level with the `bin/conf` directory.
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md]: at the same level with `bin` and `conf` directories (?? pls check the Chinese doc)

**Review by lumianph — CHANGES_REQUESTED** [review]

File: docs/en/developer/udf_develop_guide.md (hunk 9 above)
Comment (michelle-qinqin) [reply docs/en/developer/udf_develop_guide.md]: the original statment is " 需要注意的是tablet的udf目录和bin/conf目录平级，taskmanager是放到taskmanager/bin/udf目录里。"

File: docs/en/developer/udf_develop_guide.md (hunk 7 above)
Comment (michelle-qinqin) [reply docs/en/developer/udf_develop_guide.md]: The translation is finished. I am sorry I have forgoten it.

File: docs/en/developer/udf_develop_guide.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+### 2.2 Compile the Dynamic Library 
+
+- Copy the `include` directory (https://github.com/4paradigm/OpenMLDB/tree/main/include) to a certain path (like `/work/OpenMLDB/`) for later compiling. 
+- Run the compiling command. `-I` specifies the path of `include` directory. `-o` specifies the name of the dynamic library.
+
+```shell
+g++ -shared -o libtest_udf.so examples/test_udf.cc -I /work/OpenMLDB/include -std=c++11 -fPIC
+```
+
+### 2.3 Copy the Dynamic Library
```
Comment (lumianph) [inline docs/en/developer/udf_develop_guide.md:101]: please help revise the Chinese version of 2.3 accordingly.

**Review by lumianph — APPROVED** [review]

**Pull request merged**